tvb_framework/tvb/adapters/analyzers/bct_degree_adapters.py

```diff
@@ -121,81 +117,78 @@ class Strength(Degree):
+        result = bct.strengths_dir(connectivity.weights)
+        measure_index = self.build_connectivity_measure(result, connectivity, "Node instrength + outstrength")
+        return [measure_index]
 
 
 class StrengthWeights(Strength):
     """
     """
     _ui_name = "Strength and Weight"
-    _ui_description = bct_description("strengths_und_sign.m")
-    _matlab_code = "[Spos,Sneg,vpos,vneg] = strengths_und_sign(CIJ);"
+    _ui_description = bct.strengths_und_sign.__doc__
 
     def launch(self, view_model):
         connectivity = self.get_connectivity(view_model)
-        data = {'CIJ': connectivity.weights}
+        result = bct.strengths_und_sign(connectivity.weights)
 
-        result = self.execute_matlab(self._matlab_code, data=data)
-        measure_index1 = self.build_connectivity_measure(result, 'Spos', connectivity,
+        measure_index1 = self.build_connectivity_measure(result[0], connectivity,
                                                          "Nodal strength of positive weights")
-        measure_index2 = self.build_connectivity_measure(result, 'Sneg', connectivity,
+        measure_index2 = self.build_connectivity_measure(result[1], connectivity,
                                                          "Nodal strength of negative weights")
-        value1 = self.build_float_value_wrapper(result, 'vpos', "Total positive weight")
-        value2 = self.build_float_value_wrapper(result, 'vneg', "Total negative weight")
+        value1 = self.build_float_value_wrapper(result[2], "Total positive weight")
+        value2 = self.build_float_value_wrapper(result[3], "Total negative weight")
+
         return [measure_index1, measure_index2, value1, value2]
 
 
 class DensityDirected(BaseBCT):
     """
     """
     _ui_group = BCT_GROUP_DENSITY
-
     _ui_name = "Density Directed: Directed (weighted/binary) connection matrix"
-    _ui_description = bct_description("density_dir.m")
-    _matlab_code = "[kden,N,K]  = density_dir(A);"
+    _ui_description = bct.density_dir.__doc__
 
     def launch(self, view_model):
         connectivity = self.get_connectivity(view_model)
-        data = {'A': connectivity.weights}
+        result = bct.density_dir(connectivity.weights)
+
+        value1 = self.build_float_value_wrapper(result[0], title="Density")
+        value2 = self.build_int_value_wrapper(result[1], title="Number of vertices")
+        value3 = self.build_int_value_wrapper(result[2], title="Number of edges")
 
-        result = self.execute_matlab(self._matlab_code, data=data)
-        value1 = self.build_float_value_wrapper(result, 'kden', title="Density")
-        value2 = self.build_int_value_wrapper(result, 'N', title="Number of vertices")
-        value3 = self.build_int_value_wrapper(result, 'K', title="Number of edges")
         return [value1, value2, value3]
 
 
 class DensityUndirected(DensityDirected):
     """
     """
     _ui_name = "Density Unirected: Undirected (weighted/binary) connection matrix"
```

Comment:
I think there is a small typing error here. Shouldn't it be "Density Undirected:..."?